pagesections/main.statement.section.js

```diff
@@ -39,54 +39,59 @@ const MainStatementSection = ( Base ) => class extends Base {
 	 * @param {string} property
 	 * @param {string} value
 	 */
-	addMainStatement( property, value ) {
-		this.addMainStatementLink.waitForDisplayed();
-		this.addMainStatementLink.click();
+	async addMainStatement( property, value ) {
+		await this.addMainStatementLink.waitForDisplayed();
+		await this.addMainStatementLink.click();
 
-		this.mainStatementsContainer.$(
+		await this.mainStatementsContainer.$(
 			this.constructor.STATEMENT_WIDGET_SELECTORS.EDIT_INPUT_PROPERTY
 		).setValue( property );
 
-		this.selectFirstSuggestedEntityOnEntitySelector();
+		await this.selectFirstSuggestedEntityOnEntitySelector();
 
-		this.mainStatementsContainer.$(
+		await this.mainStatementsContainer.$(
 			this.constructor.STATEMENT_WIDGET_SELECTORS.EDIT_INPUT_VALUE
 		).waitForDisplayed();
-		this.mainStatementsContainer.$(
+		await this.mainStatementsContainer.$(
 			this.constructor.STATEMENT_WIDGET_SELECTORS.EDIT_INPUT_VALUE
 		).setValue( value );
 
-		this.clickSaveOnStatementElement( this.mainStatementsContainer );
+		await this.clickSaveOnStatementElement( this.mainStatementsContainer );
 
-		this.mainStatementsContainer.$(
+		await this.mainStatementsContainer.$(
 			this.constructor.STATEMENT_WIDGET_SELECTORS.EDIT_INPUT_VALUE
 		).waitForExist( { reverse: true } );
 	}
 
-	addReferenceToNthStatementOfStatementGroup( index, propertyId, referenceProperty, referenceValue ) {
-		var statement = this.getStatementElement( index, propertyId ),
-			referencesContainer = statement.$( '.wikibase-statementview-references-container' );
+	async addReferenceToNthStatementOfStatementGroup( index, propertyId, referenceProperty, referenceValue ) {
+		var statement = await this.getStatementElement( index, propertyId ),
+			referencesContainer = await statement.$( '.wikibase-statementview-references-container' );
 
-		if ( !referencesContainer.isDisplayed( this.constructor.TOOLBAR_WIDGET_SELECTORS.ADD_BUTTON ) ) {
-			statement.$( '.wikibase-statementview-references-heading' ).click();
-			statement.$( '.wikibase-statementview-references' ).waitForDisplayed();
-			referencesContainer.$( this.constructor.TOOLBAR_WIDGET_SELECTORS.ADD_BUTTON ).waitForDisplayed();
+		const containerAddButton = await referencesContainer.isDisplayed(
+			this.constructor.TOOLBAR_WIDGET_SELECTORS.ADD_BUTTON
+		);
+
+		if ( !containerAddButton ) {
```

Comment:
The variable name sounds a bit too much like an `Element` object when it's just a boolean. How about something like this? You can also just inline the variable since it's only used once.

```suggestion
		const hasAddReferenceButton = await referencesContainer.isDisplayed(
			this.constructor.TOOLBAR_WIDGET_SELECTORS.ADD_BUTTON
		);

		if ( !hasAddReferenceButton ) {
```

Reply:
Thanks. Is it okay to have the if condition like this:
```js
if ( !await referencesContainer.isDisplayed( this.constructor.TOOLBAR_WIDGET_SELECTORS.ADD_BUTTON ) ) 
```

I was a bit skeptical to have `!await`.

Reply:
Yep, that's fine!